### PR TITLE
API cleanup and doxygen

### PR DESCRIPTION
**Pull request by eile**

### CODE REVIEW

File: vmmlib/frustum.hpp [hunk 1]
```diff
@@ -33,76 +33,85 @@
...
 
 namespace vmml
 {
-template< typename T > class frustum
+/** Represents a frustum, following OpenGL conventions. */
+template< typename T > class Frustum
 {
 public:
-    T array[6]; //!< left, right, bottom, top, near, far storage
+    /** Construct a default frustum */
```
Comment (tribal-tec) [inline vmmlib/frustum.hpp]: what's a default frustum?

File: vmmlib/frustum.hpp [hunk 2]
```diff
@@ -135,177 +139,109 @@ typedef frustum< double > Frustumd;
...
+Frustum<T>::Frustum( const matrix< 4, 4, T >& projection )
 {
-    memcpy( array, source_.array, 6 * sizeof( T ) );
-    return *this;
+    _array[4] = projection( 2, 3 ) / (projection( 2, 2 ) - 1.0);
+    _array[5] = projection( 2, 3 ) / (projection( 2, 2 ) + 1.0);
+    _array[2] = nearPlane() * ( projection( 1, 2 ) - 1.0 ) / projection( 1, 1 );
+    _array[3] = nearPlane() * ( projection( 1, 2 ) + 1.0 ) / projection( 1, 1 );
+    _array[0] = nearPlane() * ( projection( 0, 2 ) - 1.0 ) / projection( 0, 0 );
+    _array[1] = nearPlane() * ( projection( 0, 2 ) + 1.0 ) / projection( 0, 0 );
```
Comment (tribal-tec) [inline vmmlib/frustum.hpp]: no farPlane in the calculation?

Comment (eile) [inline vmmlib/frustum.hpp]: l.181?

the b,t,l,r planes are specified wrt the front plane. See also adjustNear().

File: vmmlib/types.hpp [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+#endif
+
+namespace vmml
+{
+template< size_t M, size_t N, typename T > class matrix;
+template< size_t M, typename T > class vector;
+template < typename T > class quaternion;
+template< typename T > class AABB;
+template< typename T > class Frustum;
+template< class T > class FrustumCuller;
```
Comment (tribal-tec) [inline vmmlib/types.hpp]: typename vs class mixup

Comment (tribal-tec) [inline vmmlib/types.hpp]: still there

**Pull request merged**